### PR TITLE
feat: implement `SYFT_REPO_SCAN` sync type

### PR DESCRIPTION
**Pull request by patrickdevivo**
This PR should be merged _after_ #346.

`syft` (https://github.com/anchore/syft) is a CLI that generates SBOMs from various sources, including filesystems. This sync type implements a basic sync that uploads the results of `syft . -o json` on a git repo into the DB.

A view is also included with some simple JSON parsing to extract the `artifacts` list from the `syft` output JSON, for easier querying.

See here: https://github.com/anchore/syft for more context.

### CODE REVIEW

**Comment by ramiroJCB** [comment]
Could  you rebase this branch , now that  we merged  the verbose  logs  these one should only be the syft sync

File: migrations/20221002201938_syft_sync_support.up.sql [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+CREATE TABLE IF NOT EXISTS syft_repo_scans (
+    repo_id uuid PRIMARY KEY,
+    results jsonb NOT NULL
+);
+
+CREATE OR REPLACE VIEW syft_repo_artifacts AS
+SELECT
+    syft_repo_scans.repo_id,
+    a::jsonb AS artifact,
```
Comment (amenowanna) [inline migrations/20221002201938_syft_sync_support.up.sql:13]: already using the lessons we learned from the last view!

File: internal/syncer/syft_repo_scan.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+			if !errors.Is(err, pgx.ErrTxClosed) {
+				w.logger.Err(err).Msgf("could not rollback transaction")
+			}
+		}
+	}()
+
+	if _, err := tx.Exec(ctx, "DELETE FROM syft_repo_scans WHERE repo_id = $1;", j.RepoID.String()); err != nil {
+		return fmt.Errorf("exec delete: %w", err)
+	}
+
```
Comment (ramiroJCB) [inline internal/syncer/syft_repo_scan.go:67]: We should send a batch  to get the log  of the delete op :) 
`if err := w.sendBatchLogMessages(ctx, []*syncLog{{
		Type:            SyncLogTypeInfo,
		RepoSyncQueueID: j.ID,
		Message:         fmt.Sprintf("removed %d row(s) from syft_repo_scans", r.RowsAffected()),
	}}); err != nil {
		return err
	}`

File: internal/syncer/syft_repo_scan.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	if _, err := tx.Exec(ctx, "INSERT INTO syft_repo_scans (repo_id, results) VALUES ($1, $2)", j.RepoID, output); err != nil {
+		return fmt.Errorf("inserting syft results: %w", err)
+	}
+
```
Comment (ramiroJCB) [inline internal/syncer/syft_repo_scan.go:71]: Same here if err := w.sendBatchLogMessages(ctx, []*syncLog{{
		Type:            SyncLogTypeInfo,
		RepoSyncQueueID: j.ID,
		Message:         fmt.Sprintf("inserted %d row(s) into syft_repo_scans, r.RowsAffected()),
	}}); err != nil {
		return err
	}

**Review by ramiroJCB — APPROVED** [review]

**Pull request merged**